ibis/backends/polars/__init__.py

```diff
@@ -407,6 +403,7 @@ def create_view(
     def drop_table(self, name: str, *, force: bool = False) -> None:
         if name in self._tables:
             del self._tables[name]
+            self._context.unregister(name)
```

Comment:
This is a bug, so I'll break it out into a separate change to ensure it has test coverage.